src/Sarif.TestUtilities/FileDiffingTests.cs

```diff
@@ -4,15 +4,93 @@
 using System;
 using System.Globalization;
 using System.IO;
+using System.Reflection;
+using FluentAssertions;
 using Microsoft.CodeAnalysis.Sarif.Readers;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using Xunit.Abstractions;
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
     public abstract class FileDiffingTests
     {
-        protected string GenerateDiffCommand(string expected, string actual)
+        public static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        // Retrieving the source path of the tests is only used in developer ad hoc
+        // rebaselining scenarios. i.e., this path won't be consumed by AppVeyor.
+        public static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\src\Sarif.UnitTests\TestData", subdirectory));
+        }
+
+
+        private ITestOutputHelper _outputHelper;
+
+        public FileDiffingTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+
+            if (Directory.Exists(OutputFolderPath))
+            {
+                Directory.Delete(OutputFolderPath, recursive: true);
+            }
+
+            Directory.CreateDirectory(OutputFolderPath);
+        }
+
+
+        protected virtual Assembly ThisAssembly { get { return this.GetType().Assembly; } }
```

Comment:
>ThisAssembly [](start = 35, length = 12)

Consider expression body syntax:
```
protected virtual Assembly ThisAssembly => this.GetType().Assembly;
```

Reply:
yes, good place for this pattern.

---
In reply to: [226788839](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226788839) [](ancestors = 226788839)